### PR TITLE
DX-132: first draft of added functions for frontend

### PR DESCRIPTION
**Pull request by josojo**
added the function as requested.

But there is no way for solidity to have int[][] in a dynamic or predefined way in memory. Hence functions are not as powerful/informative as they could be.

### CODE REVIEW

File: contracts/DutchExchange.sol [hunk 1]
```diff
@@ -1100,7 +1100,67 @@ contract DutchExchange {
...
+         address auctionBuyToken,
+        address user
+    )
+        public
+        view
+        returns(uint[])
+    {
+        uint runningAuctionIndex = getAuctionIndex(auctionSellToken, auctionBuyToken);
+        uint[] memory sellBal = new uint[](runningAuctionIndex+1);
+        for(uint j = 0; j <= runningAuctionIndex; j++){
```
Comment (W3stside) [inline contracts/DutchExchange.sol]: A zero index pair can't be possible, so this should be `j = 1` no?

Comment (josojo) [reply contracts/DutchExchange.sol]: in js we would have to put j=1, but in solidity j=0 should be fine

Comment (W3stside) [reply contracts/DutchExchange.sol]: 👍

File: contracts/DutchExchange.sol [hunk 2]
```diff
@@ -1100,7 +1100,67 @@ contract DutchExchange {
...
+    //@param auctionSellToken are the sellTokens defining an auctionPair
+    //@param auctionBuyToken are the buyTokens defining an auctionPair
+    //@param user is the user who wants to his tokens
+    function getSellerBalanceOfCurrentAuctions(
+        address[] auctionSellToken,
+        address[] auctionBuyToken,
+        address user
+    )
+        public
+        returns(uint[])
```
Comment (Velenir) [inline contracts/DutchExchange.sol]: Should this also be a `view` function?And the next one

Comment (hacker-DOM) [reply contracts/DutchExchange.sol]: they should

File: contracts/DutchExchange.sol (hunk 1 above)
Comment (hacker-DOM) [reply contracts/DutchExchange.sol]: I agree with @W3stside that it should begin at j = 1

File: contracts/DutchExchange.sol [hunk 3]
```diff
@@ -1101,6 +1101,121 @@ contract DutchExchange {
...
+        address[] auctionSellTokens,
+        address[] auctionBuyTokens,
+        address user
+    )
+        public
+        view
+        returns (uint[])
+    {
+        uint length = auctionSellTokens.length;
+        uint length2 = auctionBuyTokens.length;
```
Comment (W3stside) [inline contracts/DutchExchange.sol:1193]: would setting `length2` here back to 0 save gas? or am I being dumb 😓

Comment (josojo) [reply contracts/DutchExchange.sol:1193]: hopefully we will never have to pay any gas for the constant view functions :)

**Pull request merged**